lib/scholar/naive_bayes/gaussian_naive_bayes.ex

```diff
@@ -0,0 +1,420 @@
+  see Stanford CS tech report STAN-CS-79-773 by Chan, Golub, and LeVeque:
+
+    http://i.stanford.edu/pub/cstr/reports/cs/tr/79/773/CS-TR-79-773.pdf
+
+  ## Options
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Returns
+
+    The function returns a struct with the following parameters:
+
+    * `:theta` - mean of each feature per class.
+
+    * `:var` - Variance of each feature per class.
+
+    * `:class_count` - number of training samples observed in each class.
+
+    * `:class_priors` - probability of each class.
+
+    * `:classes` - class labels known to the classifier.
+
+    * `:epsilon` - absolute additive value to variances.
+
+  ## Examples
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Gaussian.fit(x, y, num_classes: 3)
+      %Scholar.NaiveBayes.Gaussian{
+        theta: Nx.tensor(
+          [
+            [6.0, 7.0, 8.0],
+            [0.0, 1.0, 2.0],
+            [6.0, 7.0, 8.0]
+          ]
+        ),
+        var: Nx.tensor(
+          [
+            [1.1250000042650754e-8, 1.1250000042650754e-8, 1.1250000042650754e-8],
+            [1.1250000042650754e-8, 1.1250000042650754e-8, 1.1250000042650754e-8],
+            [9.0, 9.0, 9.0]
+          ]
+        ),
+        class_count: Nx.tensor([1.0, 1.0, 2.0]),
+        class_priors: Nx.tensor([0.25, 0.25, 0.5]),
+        classes: Nx.tensor([0, 1, 2]),
+        epsilon: Nx.tensor(1.1250000042650754e-8)
+      }
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Gaussian.fit(x, y, num_classes: 3, sample_weights: [1, 6, 2, 3])
+      %Scholar.NaiveBayes.Gaussian{
+        theta: Nx.tensor(
+          [
+            [6.0, 7.0, 8.0],
+            [0.0, 1.0, 2.0],
+            [5.0, 6.0, 7.0]
+          ]
+        ),
+        var: Nx.tensor(
+          [
+            [1.1250000042650754e-8, 1.1250000042650754e-8, 1.1250000042650754e-8],
+            [1.1250000042650754e-8, 1.1250000042650754e-8, 1.1250000042650754e-8],
+            [8.0, 8.0, 8.0]
+          ]
+          ),
+        class_count: Nx.tensor([2.0, 1.0, 9.0]),
+        class_priors: Nx.tensor([0.1666666716337204, 0.0833333358168602, 0.75]),
+        classes: Nx.tensor([0, 1, 2]),
+        epsilon: Nx.tensor(1.1250000042650754e-8)
+      }
+  """
+  deftransform fit(x, y, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    input_rank = Nx.rank(x)
+    targets_rank = Nx.rank(y)
+
+    if input_rank != 2 or targets_rank != 1 do
+      raise ArgumentError,
+            "Wrong input rank. Expected: 2 for x and 1 for y, got: #{inspect(input_rank)} for x and #{inspect(targets_rank)} for y"
+    end
+
+    {num_samples, _} = Nx.shape(x)
+    {num_targets} = Nx.shape(y)
+
+    if num_samples != num_targets do
+      raise ArgumentError,
+            "Wrong input shape. Expect x to have the same first dimension as y, got: #{inspect(num_samples)} for x and #{inspect(num_targets)} for y"
```

Comment:
No need for inspect for numbers and small grammar fix:

```suggestion
            "wrong input shape. Expected x to have the same first dimension as y, got: #{num_samples} for x and #{num_targets} for y"
```